src/analysis/complex/upper_half_plane.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright (c) 2021 Alex Kontorovich and Heather Macbeth and Marc Masdeu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Kontorovich, Heather Macbeth, Marc Masdeu
+-/
+
+import linear_algebra.special_linear_group
+import analysis.complex.basic
+import group_theory.group_action.defs
+
+/-!
+# The upper half plane and its automorphisms
+
+This file defines `upper_half_plane` to be the upper half plane in `ℂ`.
+
+We furthermore equip it with the structure of an `SL(2,ℝ)` action by
+fractional linear transformations.
+
+We define the notation `ℍ` for the upper half plane available in the locale
+`upper_half_plane` so as not to conflict with the quaternions.
+-/
+
+noncomputable theory
+
+open matrix matrix.special_linear_group
+
+open_locale classical big_operators matrix_groups
+
+local attribute [instance] fintype.card_fin_even
+
+/-- The upper half plane -/
```

Comment:
```suggestion
/-- The open upper half plane -/
```
I think it is useful to indicate in the docstring that the real line is not part of the upper half plane. This can be done using the word `open`, but feel free to do it otherwise.